### ISSUE TITLE
Guide shows behind untitled editor on first run

### ISSUE THREAD

**Issue by kevinsawicki** [atom/welcome#26]
Wasn't sure if this was intentional or not, but it felt weird to have the guide behind the untitled buffer instead of in front of it when Atom is first launched.

![screen shot 2015-03-02 at 9 04 06 am](https://cloud.githubusercontent.com/assets/671378/6445771/26097ed6-c0bb-11e4-8ad9-53dca7fae158.png)

**Comment by benogle** [comment]
It's just that the guide is focused, correct? Likely because it is opened last.

**Comment by kevinsawicki** [comment]
The editor was focused in this case I believe.

**Comment by benogle** [comment]
I guess I'm confused. Can you give me exact steps? My assumption:

* open welcome
* it opens welcome in left pane then guide in right pane; at this point the right pane has focus
* cmd+n
* opens new buffer in right pane

**Comment by kevinsawicki** [comment]
>  Can you give me exact steps?

* Open Atom for the first time
* That screenshot is what I saw

**Comment by benogle** [comment]
Here is [the beginning][begin] of the open-initial-empty-buffer code, and here is [the handler][end] in the renderer process. The entire way through, this initial buffer opening is implicit via a null path. My thoughts right now are: 

* I would like this to be explicit for clarity. It wasnt very easy to follow through main -> app -> window -> IPC.
* I want to get metadata through to the [renderer handler][end] indicating that the buffer is or is not the initial empty buffer
* I want to have an event like `atom.onWillOpenInitialEmptyBuffer({cancel})` where `cancel` will allow a package to _not_ open said buffer.

Any opinions on this approach? @kevinsawicki @nathansobo?

[begin]:https://github.com/atom/atom/blob/8fac0e87788aaf398adee0bb75f034ef497eb971/src/browser/atom-application.coffee#L93
[end]:https://github.com/atom/atom/blob/8fac0e87788aaf398adee0bb75f034ef497eb971/src/window-event-handler.coffee#L32-L33

**Comment by nathansobo** [comment]
I don't have a ton of context on this but it seems reasonable. So to clarify, the issue is that we're opening an initial empty buffer when we want to only show the welcome screen and you're seeking to preempt it?

**Comment by benogle** [comment]
> when we want to only show the welcome screen and you're seeking to preempt it

Yeah

**Comment by nathansobo** [comment]
Sounds good.

**Comment by kevinsawicki** [comment]
> Any opinions on this approach?

> I would like this to be explicit for clarity.

:+1: This would be great, this code path is very hard to follow currently and easy to break on the browser process side.

> I want to get metadata through to the renderer handler indicating that the buffer is or is not the initial empty buffer

It would be great if we could get this to be 100% a render process concern

> I want to have an event like atom.onWillOpenInitialEmptyBuffer({cancel})

I'm down for an API, though I think having the untitled buffer behind the welcome guide is kind of nice, once you digest the welcome information and close it, you got a buffer ready to go.

**Comment by benogle** [comment]
> I think having the untitled buffer behind the welcome guide is kind of nice, once you digest the welcome information and close it, you got a buffer ready to go.

The only way to do that today without changing browser code is to add the welcome panes on a long timeout (> 500ms).

Check out the PR at https://github.com/atom/atom/pull/5812. I think it would be useful to pass some information indicating whether it's opening the initial buffer. It has upsides elsewhere too: [This person](https://github.com/atom/atom/issues/5820) could disable it in their init script, we could make an option for it, etc.
